File: problems/java/12. Integer to Roman.java
class Solution {
    public String intToRoman(int num) {
        int[] vals = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] syms = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        
        StringBuilder res = new StringBuilder();
        
        for (int i = 0; i < vals.length; i++) {
            while (num >= vals[i]) {
                res.append(syms[i]);
                num -= vals[i];
            }
        }
        
        return res.toString();
    }
}

File: problems/C++/12. Integer to Roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<int> vals = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        vector<string> syms = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        
        string res = "";
        
        for (int i = 0; i < vals.size(); ++i) {
            while (num >= vals[i]) {
                res += syms[i];
                num -= vals[i];
            }
        }
        
        return res;
    }
};